library/src/scala/tasty/util/ShowSourceCode.scala

```diff
@@ -89,33 +89,39 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
           case _ => true
         }
         if (parents1.nonEmpty) {
-          sb.append(" extends")
-          parents1.foreach {
-            case parent@Term.Apply(Term.TypeApply(Term.Select(Term.New(tpt), _, _), targs), args) =>
-              this += " "
-              printTypeTree(tpt)
+          this += " extends "
+
+          def printParent(parent: Parent): Unit = parent match {
+            case parent @ Term.TypeApply(fun, targs) =>
+              printParent(fun)
               this += "["
               printTypeOrBoundsTrees(targs, ", ")
               this += "]"
-              if (args.nonEmpty) {
-                this += "("
-                printTrees(args, ", ")
-                this += ")"
-              }
 
-            case parent@Term.Apply(Term.Select(Term.New(tpt), _, _), args) =>
-              this += " "
+            case parent @ Term.Apply(fun, args) =>
+              printParent(fun)
+              this += "("
+              printTrees(args, ", ")
+              this += ")"
+
+            case parent @ Term.Select(Term.New(tpt), _, _) =>
               printTypeTree(tpt)
-              if (args.nonEmpty) {
-                this += "("
-                printTrees(args, ", ")
-                this += ")"
-              }
 
-            case parent@TypeTree() =>
-              sb.append(" ")
+            case parent @ TypeTree() =>
               printTypeTree(parent)
+
+            case parent @ Term() => throw new MatchError(parent.show)
+          }
+
+          def printSeparated(list: List[Parent]): Unit = list match {
+            case Nil =>
```

Comment:
unreachable case, throw an exception here if reached?

Reply:
I moved the `nonEmpty` check, now this is reachable.